internal/certs/pool.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package certs
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+)
+
+// PoolWithCACertificate returns a new pool that includes the CA certificate
+// in the given path.
+func PoolWithCACertificate(path string) (*x509.CertPool, error) {
+	pool := x509.NewCertPool()
+	err := addCACertificateToPool(pool, path)
+	if err != nil {
+		return nil, err
+	}
+	return pool, nil
+}
```

Comment:
**question** Currently, it is just being used the function related to the system certs pool. Should this one be kept too? It's true that it could allow us to test just a subset of CAs in case it is needed easily.

Reply:
In https://github.com/elastic/elastic-package/pull/789 I will also use it to [check](https://github.com/elastic/elastic-package/pull/789/files#diff-5f4340f6bec3f455c2fdb1aed7ab1ae1fd23cc179d8520cdbf275461794c504eR133) that the CA in the profile verifies the service certificates.

I can remove it by now, it may happen that we don't use it at the end, depending on the other PR :slightly_smiling_face: